packages/react/src/components/Combobox/Combobox.test.tsx

```diff
@@ -0,0 +1,914 @@
+    </Combobox>
+  );
+  const combobox = screen.getByRole('combobox');
+
+  expect(screen.getByRole('combobox').getAttribute('value')).toEqual('Banana');
+
+  fireEvent.focus(combobox);
+  assertListboxIsOpen(true);
+  assertOptionIsSelected(1);
+});
+
+test('should set selected value with "value" prop', () => {
+  render(
+    <Combobox label="label" value="Banana">
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+  const combobox = screen.getByRole('combobox');
+
+  expect(screen.getByRole('combobox').getAttribute('value')).toEqual('Banana');
+
+  fireEvent.focus(combobox);
+  assertListboxIsOpen(true);
+  assertOptionIsSelected(1);
+});
+
+test('should not render hidden input when name is not provided', () => {
+  render(
+    <Combobox label="label" value="Banana">
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+
+  expect(document.querySelector('input[type="hidden"]')).toBeFalsy();
+});
+
+test('should render hidden input with value from text contents of ComboboxOption', () => {
+  render(
+    <Combobox label="label" name="fruit" value="Banana">
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+
+  expect(
+    document.querySelector('input[type="hidden"]')?.getAttribute('value')
+  ).toEqual('Banana');
+});
+
+test('should render hidden input with value from value from ComboboxOption', () => {
+  render(
+    <Combobox label="label" name="fruit" value="Banana">
+      <ComboboxOption value="Apple">🍎</ComboboxOption>
+      <ComboboxOption value="Banana">🍌</ComboboxOption>
+      <ComboboxOption value="Cantaloupe">🍈</ComboboxOption>
+    </Combobox>
+  );
+
+  expect(
+    document.querySelector('input[type="hidden"]')?.getAttribute('value')
+  ).toEqual('Banana');
+});
+
+test('should render hidden input with value from formValue from ComboboxOption', () => {
+  render(
+    <Combobox label="label" name="fruit" value="Banana">
+      <ComboboxOption formValue="1">Apple</ComboboxOption>
+      <ComboboxOption formValue="2">Banana</ComboboxOption>
+      <ComboboxOption formValue="3">Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+
+  expect(
+    document.querySelector('input[type="hidden"]')?.getAttribute('value')
+  ).toEqual('2');
+});
+
+test('should support portal element for combobox listbox', () => {
+  const element = document.createElement('div');
+  render(
+    <Combobox label="label" portal={element}>
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+  expect(
+    element.contains(element.querySelector('[role="listbox"]'))
+  ).toBeTruthy();
```

Comment:
You might be looking for [within](https://testing-library.com/docs/dom-testing-library/api-within/) here, which would allow you to use RTL's selectors